api/wasm.go

```diff
@@ -329,6 +329,8 @@ type Function interface {
 	// WithCloseOnContextDone on wazero.RuntimeConfig for detail. See examples in context_done_example_test.go for
 	// the end-to-end demonstrations of how these terminations can be performed.
 	Call(ctx context.Context, params ...uint64) ([]uint64, error)
+
+	private()
```

Comment:
so I think maybe I understand this. Basically, this doesn't stop external implementation, rather it slows it. So people need to do embed as we do in other parts of the PR.

If others aren't against this approach, I'm fine with it. We'd boilerplate docs. We can even be more explicit in the name if we want!

```suggestion
	// wazeroOnly means this should only be implemented by wazero.
	wazeroOnly()
```

Reply:
Exactly, agree, and `wazeroOnly` is a good name.